### PR TITLE
Fix JSON serialization when working with bson primitives

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: app/worker/utils.go [hunk 1]
```diff
@@ -146,6 +147,36 @@ func processJobJSON(data interface{}, workflow *model.Workflow, job *model.Job)
 		return result
 	} else if value, ok := data.(string); ok {
```
Comment (alfrunes) [inline app/worker/utils.go:148]: I prefer using type-switches
```
switch value := data.(type) {
	case string:
		return processJobString(value, workflow, job)
	...
}
```

File: app/worker/utils.go [hunk 2]
```diff
@@ -146,6 +147,36 @@ func processJobJSON(data interface{}, workflow *model.Workflow, job *model.Job)
...
+		for key, item := range value {
+			result[key] = processJobJSON(item, workflow, job)
+		}
+		return result
+	} else if value, ok := data.([]primitive.M); ok {
+		result := make([]interface{}, len(value))
+		for i, item := range value {
+			result[i] = processJobJSON(item, workflow, job)
+		}
+		return result
```
Comment (alfrunes) [inline app/worker/utils.go:173]: I believe these are impossible cases:
> 5. When unmarshalling, a field of type interface{} will follow the D/M type mappings listed above. BSON documents unmarshalled into an interface{} field will be unmarshalled as a D.
https://pkg.go.dev/go.mongodb.org/mongo-driver/bson@v1.4.3#hdr-Structs

**Review by alfrunes — APPROVED** [review]
LGTM

I just have a couple less important nitpicks.

File: app/worker/utils.go (hunk 2 above)
Comment (tranchitella) [reply app/worker/utils.go]: Ok, leaving them out and focusing only on primitive.D

File: app/worker/utils.go (hunk 1 above)
Comment (tranchitella) [reply app/worker/utils.go]: Way nicer, thanks

**Review by alfrunes — APPROVED** [review]

**Pull request merged**